### PR TITLE
Feat/share photo video backend #122

### PR DESCRIPTION
**Pull request by samgo1**
### Share photo/video feature

- click on the share toggle on top bar
- it will become gold
- take a picture or video
- choose the sms app to send it through text message

I tested it with the email app because I don't have a SIM card in my test phone

### CODE REVIEW

**Comment by lkadian** [comment]
Tested it out on my phone and it works perfectly for photos and videos, nice job!

edit: One small thing I noticed is that when you take a picture with a filter and share applied (e.g. negative filter), then come back to the original camera view, it will not have a filter showing, but if you take another picture there will still be that negative filter applied. I don't think this is related to what you did though, more a general case that when we resume from another activity the filter will disappear from the view but still be applied to the final photo. I will look into it

File: app/src/main/kotlin/com/simplemobiletools/camera/implementations/QRScanner.java [hunk 1]
```diff
@@ -56,7 +56,7 @@
     private AlertDialog alert = null;
 
     /** Scheduling flags */
-    public static boolean qr_requested = true;
+    public static boolean qr_requested = false;
```
Comment (lkadian) [inline app/src/main/kotlin/com/simplemobiletools/camera/implementations/QRScanner.java:59]: any reason for changing this? Not so familiar with the QR scanner code so just wondering.

Comment (samgo1) [reply app/src/main/kotlin/com/simplemobiletools/camera/implementations/QRScanner.java:59]: yes in PhotoProcessor.kt, we have this code when attempting to save a picture
Screen Shot 2019-04-06 at 10 44 01 AM
when the _first picture_ gets taken it gets in that conditional even when we didn't request the scanner.
I'm pretty sure I checked everything and it was safe to set it to false initially, correct me if I'm wrong @aa-software2112

**Comment by samgo1** [comment]
> Tested it out on my phone and it works perfectly for photos and videos, nice job!
> 
> edit: One small thing I noticed is that when you take a picture with a filter (e.g. negative filter), then come back to the original camera view, it will not have a filter showing, but if you take another picture there will still be that negative filter applied.

Does this happens only when the share toggle is on?

**Comment by lkadian** [comment]
> Does this happens only when the share toggle is on?

After testing again I realized this is a general issue and not related to your PR at all. Here is the issue:

1. Apply a filter (e.g. mono), don't need to take a picture
2. Go to any other activity (e.g. settings or share activity)
3. Come back to main activity and the filter will not be applied to the Camera view, but if you take a picture, it will actually apply the filter you had in (1) to the picture.

This is related to the filters implementation and #138 , I will make an issue for it

**Review by lkadian — APPROVED** [review]

File: app/src/main/kotlin/com/simplemobiletools/camera/implementations/QRScanner.java (hunk 1 above)
Comment (aa-software2112) [reply app/src/main/kotlin/com/simplemobiletools/camera/implementations/QRScanner.java:59]: that's odd, it should be false to begin with; you're right

**Comment by aa-software2112** [comment]
> > Does this happens only when the share toggle is on?
> 
> After testing again I realized this is a general issue and not related to your PR at all. Here is the issue:
> 
> 1. Apply a filter (e.g. mono), don't need to take a picture
> 2. Go to any other activity (e.g. settings or share activity)
> 3. Come back to main activity and the filter will not be applied to the Camera view, but if you take a picture, it will actually apply the filter you had in (1) to the picture.
> 
> This is related to the filters implementation and #138 , I will make an issue for it
Please make this issue and reference @Lussier115, he is likely the best person for this

**Comment by lkadian** [comment]
> > > Does this happens only when the share toggle is on?
> > 
> > 
> > After testing again I realized this is a general issue and not related to your PR at all. Here is the issue:
> > 
> > 1. Apply a filter (e.g. mono), don't need to take a picture
> > 2. Go to any other activity (e.g. settings or share activity)
> > 3. Come back to main activity and the filter will not be applied to the Camera view, but if you take a picture, it will actually apply the filter you had in (1) to the picture.
> > 
> > This is related to the filters implementation and #138 , I will make an issue for it
> > Please make this issue and reference @Lussier115, he is likely the best person for this

I did and assigned myself, I should be able to fix it

**Pull request merged**